internal/dag/gatewayapi_processor.go

```diff
@@ -405,6 +417,47 @@ func (p *GatewayAPIProcessor) computeHTTPRoute(route *gatewayapi_v1alpha1.HTTPRo
 	}
 }
 
+func pathMatchCondition(mc *matchConditions, match *gatewayapi_v1alpha1.HTTPPathMatch) error {
+
+	if match == nil {
+		mc.pathMatchConditions = append(mc.pathMatchConditions, &PrefixMatchCondition{Prefix: "/"})
+		return nil
+	}
+
+	if match.Type == nil {
+		// If path match type is not defined, default to 'PrefixMatch'.
+		mc.pathMatchConditions = append(mc.pathMatchConditions, &PrefixMatchCondition{Prefix: pointer.StringDeref(match.Value, "/")})
+	} else {
+		switch *match.Type {
+		case gatewayapi_v1alpha1.PathMatchPrefix:
+			mc.pathMatchConditions = append(mc.pathMatchConditions, &PrefixMatchCondition{Prefix: pointer.StringDeref(match.Value, "/")})
+		case gatewayapi_v1alpha1.PathMatchExact:
+			mc.pathMatchConditions = append(mc.pathMatchConditions, &ExactMatchCondition{Path: pointer.StringDeref(match.Value, "/")})
+		default:
+			return fmt.Errorf("HTTPRoute.Spec.Rules.PathMatch: Only Prefix match type and Exact match type are supported")
+		}
+	}
+	return nil
+}
+
+func headerMatchCondition(mc *matchConditions, match *gatewayapi_v1alpha1.HTTPHeaderMatch) error {
+	if match == nil {
+		return nil
+	}
+
+	if match.Type != nil {
```

Comment:
Looks like this is being defaulted to Exact: https://github.com/kubernetes-sigs/gateway-api/blob/cbbdc4c1ba2b0468f5412477c0de2f02d4089abf/apis/v1alpha1/httproute_types.go#L315

👍🏽

Reply:
Gotcha, I updated. We're finding some good bugs in this PR. =)